### PR TITLE
Grammar fixes - British mode enabled

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes various grammar & vocabulary fixes across scapy using `codespell`.
- adds some grammar spelling regression tests

Only 2 variables are actually modified:
- `endianess` -> `endianness` (from the very very recent PRs, it's ok to change it right now)
- `additionnal` -> `additional` (from StrStopField. It is the less-used field in entire scapy, and the files using it are not even calling it by its name, but as arguments)

**_Might be annoying for history :/ feel free to tell me if that's a change too big._**

### CODE REVIEW

File: scapy/contrib/gtp_v2.py [hunk 1]
```diff
@@ -464,7 +464,7 @@ class PCO_Secondary_NBNS(PCO_Option):
     0x000c: 'P-CSCF IPv4 Address Request',
     0x0010: 'IPv4 Link MTU Request',
     0x8021: 'IPCP',
-    0xc023: 'Password Authentification Protocol',
```
Comment (gpotter2) [inline scapy/contrib/gtp_v2.py]: I looked up that one, and it seems correct: Authentification is French, Authentication seems to be correct

**Comment by p-l-** [comment]
I'm OK with this, but I'd like to have a minimal test added so that we don't have to merge such PRs every now and then. Maybe a simple `grep` to test for frequent mistakes would be enough?

**Comment by gpotter2** [comment]
That’s what “codespell” does (github project), we could try to integrate a test using it.

It has a few issues, with “ans” (answer in scapy), which it tries to turn to “and”, so we should make this one ignored, and “see secdev [...] for informations » (it should be information with no s, but I don’t want to replace every file in scapy just for this)

**Comment by p-l-** [comment]
if we add a test, I'm in favour of changing informations to information everywhere.

**Comment by gpotter2** [comment]
Added a test

**Pull request merged**